enhancements/sig-architecture/8-addon-framework/README.md

```diff
@@ -0,0 +1,200 @@
+### User Stories
+
+#### Story 1
+
+As an open-cluster-management admin on hub, I can add an addon into the hub. For example, I want to install the submariner addon into the hub.
+
+#### Story 2
+
+As an open-cluster-management admin on hub, I can enable the addon installation on all managedclusters.
+
+#### Story 3
+
+As an admin on a managed cluster, I can enable and config the addon agent on a certain managed cluster. I was able to enable the addon agent on hub, which triggers the agent installation on the managed cluster.
+
+#### Story 4
+
+As an addon developer, I can easily develop the addon using the library without deep understanding on how agent manifests are distributed and registerd to hub.
+
+
+### Standardize addon procedure
+
+#### Register an addon on hub
+
+A common process for an addon to be installed on the hub cluster follows the procedure below:
+
+1. Cluster admin on hub deploy a certain addon controller
+2. Addon controller registers itself to the hub by creating a ClusterManagerAddon resource on hub
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddon
+metadata:
+ name: submariner
+spec:
+ addonMeta:
+   displayName: submariner
+   description: submariner-addon
+ addonConfiguration:
+   crd: submarineraddonconfig.addon.open-cluster-management.io
+   name: submariner-config
+```
+
+#### Enable addon agent on a managed cluster
+
+1. A user creates a ManagedClusterAddon resource of which the name is the same as the name of ClusterManagerAddon resource in the cluster namespace.
+```yaml
+ apiVersion: addon.open-cluster-management.io/v1alpha1
+ kind: ManagedClusterAddon
+ metadata:
+   name: submariner
+   namespace: cluster1
+```
+2. Addon controller on hub deploys the agent by creating manifestworks in cluster namespace.
+3. The agent is started on a managed cluster, and registers back using credential.
+
+#### Communication pattern
+
+An addon agent in a managed cluster will communicate with the hub:
+- The addon agent talks to the kube apiserver of the hub.  For reading desired state and for pushing small quantities of data that are stored in kubernetes API objects, this is the recommended path.
+- The addon agent talks to another endpoint on the hub.  For writing large amounts of data to the hub or sending data that doesn’t fit cleanly into a kubernetes style API, this is the recommended approach.
+
+In both ways, the communication should be properly authenticated/authorized.
+
+##### mTLS for authentication
+
+The addon agent should be able to register to the hub by issuing a csr request and choose the signer. If the addon agent talks to the kube-apiserver of the hub, the agent should use the `kubernetes.io/kube-apiserver-client` to create the csr. Otherwise, the addon agent should use a customized signer for different endpoints. The endpoint, which supposedly belongs to the hub component of the addon should be able to accept the cert signed by the customized signer for this addon.
```

Comment:
nit: line break on sentences for commenting ability.

Reply:
updated